tests/core/training/test_interactive.py

```diff
@@ -624,3 +624,32 @@ async def test_not_getting_trackers_when_skipping_visualization(
         )
 
     get_trackers.assert_not_called()
+
+
+async def test_retry_on_error(monkeypatch: MonkeyPatch):
+    def func(path, x):
+        if x:
+            return
+        else:
+            raise PermissionError()
+
+    class QuestionaryConfirmMock:
+        def __init__(self, tries):
+            self.tries = tries
+
+        def __call__(self, text):
+            return self
+
+        def ask(self):
+            self.tries -= 1
+            if self.tries == 0:
+                return False
+            else:
+                return True
+
+    interactive.questionary.confirm = QuestionaryConfirmMock(3)
+
+    await interactive._retry_on_error(func, "export_path", True)
```

Comment:
optional: use a lambda for `func`. I think it's more clear what you're testing then (currently I have to follow the bool parameter and then see what `func` is supposed to do in that case.

Reply:
I think lambdas would be clearer to read as well. However it is weird to raise exceptions from lambdas (lambda x: raise y) doesn't work and also argument passing is tested this way as well.

Reply:
optional: you could do something like this. Using the mock you could also test the passing of the args. 
```suggestion
    await interactive._retry_on_error(Mock(side_effect=PermissionError()), "export_path")
```